controller/src/main/java/io/pravega/controller/server/bucket/PeriodicWatermarking.java

```diff
@@ -387,14 +391,28 @@ boolean checkExistsInCache(Stream stream) {
         private final ConcurrentHashMap<String, Long> inactiveWriters;
         
         WatermarkClient(Stream stream, ClientConfig clientConfig) {
-            this(stream, SynchronizerClientFactory.withScope(stream.getScope(), clientConfig));
+            this(stream, SynchronizerClientFactory.withScope(stream.getScope(), clientConfig), true);
```

Comment:
Can we avoid creating a new SynchronizerClientFactory for every WatermarkClient? Every factory creates an new thread pool.